module/zfs/spa.c

```diff
@@ -1669,23 +1707,95 @@ spa_thread(void *arg)
 }
 #endif
 
-extern metaslab_ops_t *metaslab_allocator(spa_t *spa);
+/*
+ * Returns with the spa_chain_map_lock held. This prevents the shared log
+ * pool from being exported or deleted while a pool is being activated that
+ * depends on it.
+ */
+static int
+get_shared_log_pool(nvlist_t *config, spa_t **out)
+{
+	if (config == 0)
```

Comment:
Cosmetics, but do we really need to check the argument here, and if really so, why `0`, not `NULL`?

Reply:
I'm pretty sure we do actually need to check the argument, but I'll use NULL. There are a *lot* of weird edgecases like this in the zpool import/create process that I would also not have thought were possible.